network/src/web.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  UnsupportedBrowserException,
+  WebPlugin,
+  PluginListenerHandle,
+} from '@capacitor/core';
+import {
+  NetworkPlugin,
+  NetworkStatus,
+  NetworkStatusConnectionType,
+} from './definitions';
+
+declare var window: any;
+
+function translateConnection(connection: any): NetworkStatusConnectionType {
+  let result: NetworkStatusConnectionType = 'unknown';
+  const type = connection ? connection.type || connection.effectiveType : null;
+  if (type && typeof type === 'string') {
+    switch (type) {
+      // possible type values
+      case 'bluetooth':
+      case 'cellular':
+        result = 'cellular';
+        break;
+      case 'none':
+        result = 'none';
+        break;
+      case 'ethernet':
+      case 'wifi':
+      case 'wimax':
+        result = 'wifi';
+        break;
+      case 'other':
+      case 'unknown':
+        result = 'unknown';
+        break;
+      // possible effectiveType values
+      case 'slow-2g':
+      case '2g':
+      case '3g':
+        result = 'cellular';
+        break;
+      case '4g':
+        result = 'wifi';
+        break;
+      default:
+        break;
+    }
+  }
+  return result;
+}
+
+export class NetworkWeb extends WebPlugin implements NetworkPlugin {
+  constructor() {
+    super({ name: 'Network' });
+  }
+
+  async getStatus(): Promise<NetworkStatus> {
+    if (!window.navigator) {
+      throw new UnsupportedBrowserException(
+        'Browser does not support the Network Information API',
+      );
+    }
```

Comment:
window.navigator is always defined, maybe it's better to just check if onLine is defined in navigator

```suggestion
    if ('onLine' in navigator) {
      throw new UnsupportedBrowserException(
        'Browser does not support the Network Information API',
      );
    }
```

But onLine has been available for a long long time, so maybe we can even remove the check
https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine

Reply:
After accepting this change, I have reverted it. `('onLine' in navigator)` would always fail and report that the API was not available.